src/Workspaces/Core/Portable/Progress/CodeAnalysisProgress.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using Microsoft.CodeAnalysis.Progress;
+
+namespace Microsoft.CodeAnalysis;
+
+/// <summary>
+/// Represents the progress of an operation.  Commonly used to update a UI visible to a user when a long running
+/// operation is happening.
+/// </summary>
+public sealed class CodeAnalysisProgress
+{
+    /// <summary>
+    /// Used when bridging from an api that does not show progress to the user to an api that can update progress if
+    /// available.  This should be used sparingly.  Locations that currently do not show progress should ideally be
+    /// migrated to ones that do so that long running operations are visible to the user in a coherent fashion.
+    /// </summary>
+    internal static readonly IProgress<CodeAnalysisProgress> None = NullProgress<CodeAnalysisProgress>.Instance;
+
+    internal bool ClearValue { get; init; }
+    internal int? CompleteItemValue { get; init; }
+    internal int? IncompleteItemsValue { get; init; }
+    internal string? DescriptionValue { get; init; }
+
+    /// <summary>
+    /// Updates the UI showing the progress of the current operation to the specified <paramref name="description"/>.
```

Comment:
This sounds a bit odd to me. Maybe I'm being picky, but this doesn't actually update the UI, but rather creates a representation that indicates a description is updated.

Reply:
updated for clarity.